src/lib.rs

```diff
@@ -301,12 +301,44 @@ fn option<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Node,
     Ok((input, Node::Optional(Box::new(alternation))))
 }
 
+/// ```abnf
+/// char-val = case-insensitive-string / case-sensitive-string
+/// ```
+fn char_val<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&str, StringLiteral, E> {
+    alt((
+        map(case_insensitive_string, |str| {
+            StringLiteral::case_insensitive(str.to_owned())
+        }),
+        map(case_sensitive_string, |str| {
+            StringLiteral::case_sensitive(str.to_owned())
+        }),
+    ))(input)
+}
+
+/// ```abnf
+/// case-insensitive-string = [ "%i" ] quoted-string
+/// ```
+fn case_insensitive_string<'a, E: ParseError<&'a str>>(
+    input: &'a str,
+) -> IResult<&'a str, &str, E> {
+    let marker = preceded(char('%'), alt((char('i'), char('I'))));
```

Comment:
Hehe, seems you cared about everything! :-) I would not have cared about "%I" and "%S". But... it's what the RFC tells us to do, so ... :+1: